### PR TITLE
Improve tracking of autobuild errors

### PR DESCRIPTION
**Pull request by henrymercer**
Improve the way we surface errors while automatically attempting to build code, particularly in circumstances where the autobuilder doesn't produce diagnostic information.

Specifically, extract just the `[ERROR]` part of the stderr stream, and link users to the troubleshooting page in the docs.

Also mark Maven and Gradle build failures as configuration errors.

### Merge / deployment checklist

- [ ] Confirm this change is backwards compatible with existing workflows.
- [ ] Confirm the [readme](https://github.com/github/codeql-action/blob/main/README.md) has been updated if necessary.
- [ ] Confirm the [changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md) has been updated if necessary.

### CODE REVIEW

File: src/cli-errors.ts [hunk 1]
```diff
@@ -145,6 +165,11 @@ export const cliErrorsConfig: Record<
   [CliConfigErrorCategory.InvalidSourceRoot]: {
     cliErrorMessageCandidates: [new RegExp("Invalid source root")],
   },
+  [CliConfigErrorCategory.MavenBuildFailed]: {
+    cliErrorMessageCandidates: [
+      new RegExp("[autobuild] [ERROR] Failed to execute goal"),
```
Comment (nickfyson) [inline src/cli-errors.ts:170]: I haven't had to wrangle Maven in many a year now, but I take it this is an extremely characteristic string? 😆

File: src/cli-errors.ts [hunk 2]
```diff
@@ -96,6 +107,14 @@ function extractFatalErrors(error: string): string | undefined {
   return undefined;
 }
 
+function extractAutobuildErrors(error: string): string | undefined {
+  const pattern = /.*\[autobuild\] \[ERROR\] (.*)/gi;
+  return (
+    [...error.matchAll(pattern)].map((match) => match[1]).join("\n") ||
```
Comment (nickfyson) [inline src/cli-errors.ts:113]: Is there any value in bounding the max number of lines we'll return here, or given they're generated by our own script are we confident it'll never be _too_ outlandish...? 🙂

**Review by nickfyson — APPROVED** [review]
LGTM

File: src/cli-errors.ts (hunk 1 above)
Comment (henrymercer) [reply src/cli-errors.ts:170]: I think this is fairly characteristic :)

File: src/cli-errors.ts (hunk 2 above)
Comment (henrymercer) [reply src/cli-errors.ts:113]: Good point — we later cap what we send to telemetry, but I think it would probably make sense to add a cap here too.

**Review by nickfyson — APPROVED** [review]

**Review by nickfyson — APPROVED** [review]

**Pull request merged**